nl2sql/llms/vertexai.py

```diff
@@ -17,71 +17,66 @@
 """
 from google.cloud import aiplatform_v1beta1
 from google.protobuf import struct_pb2
-from langchain.llms.vertexai import VertexAI
-
+from langchain_google_vertexai import VertexAI
+from typing import ClassVar
 
 class ExtendedVertexAI(VertexAI):
     """
     Adds utility functions to GooglePalm
     """
+    token_limits: ClassVar[dict[str, int]] = {
+        "text-bison" : 2048,
+        "text-bison-32k": 8000,
+        "gemini-1.5-flash-001": 8000,
+        "gemini-1.0-pro-002": 8000,
+        "gemini-1.5-pro-001": 8000,
+    }
 
     def get_num_tokens(self, text: str) -> int:
         """
         Returns the token count for some text
         """
-        token_struct = struct_pb2.Struct()
-        token_struct.update({"content": text})
-        return (
-            aiplatform_v1beta1.PredictionServiceClient(
-                client_options={
-                    "api_endpoint": f"{self.location}-aiplatform.googleapis.com"
-                }
-            )
-            .count_tokens(
-                endpoint=self.client._endpoint_name,  # pylint: disable = protected-access
-                instances=[struct_pb2.Value(struct_value=token_struct)],
+        if self.model_name.startswith("gemini"):
+            return self.client.count_tokens(text).total_tokens
+        else:
+            token_struct = struct_pb2.Struct()
+            token_struct.update({"content": text})
+            return (
+                aiplatform_v1beta1.PredictionServiceClient(
+                    client_options={
+                        "api_endpoint": f"{self.location}-aiplatform.googleapis.com"
+                    }
+                )
+                .count_tokens(
+                    endpoint=self.client._endpoint_name,  # pylint: disable = protected-access
+                    instances=[struct_pb2.Value(struct_value=token_struct)],
+                )
+                .total_tokens
             )
-            .total_tokens
-        )
 
     def get_max_input_tokens(self) -> int:
         """
         Returns the maximum number of input tokens allowed
         """
-        if self.metadata:
-            return self.metadata["max_input_tokens"]
-        raise ValueError("LLM initialized without max_input_tokens")
+        raise NotImplementedError
```

Comment:
Can we return this from the `token_limits` dictionary, and raise an NotImplemementedError if the model name is not present there?